### ISSUE TITLE
[HOLD for payment 2023-09-04] [$1000] Web - Money request - Copy to clipboard button does not copy content of an IOU edit comment

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#24569]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Make money request from account A to account B
2. Click on the IOU preview component twice
3. Click on the amount and change it
4. Hover over the posted comment and click on copy to clipboard
5. Paste the content elsewhere

## Expected Result:
The content should be copied when clicking on the copy to clipboard icon

## Actual Result:
Nothing gets copied when clicking the copy to clipboard icon

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** v1.3.54-7

**Reproducible in staging?:** Y

**Reproducible in production?:** N

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/115492554/ae1cbb4a-f3b0-4a0f-b13d-2bfcc4452e5e

https://github.com/Expensify/App/assets/115492554/47abd170-2f25-4a64-adb5-d28821c03451

**Expensify/Expensify Issue URL:**

**Issue reported by:** @Nathan-Mulugeta

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1692045636871669

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~015fd8284ce80abac6</li>
        <li>Upwork Job ID: 1691425057366032384</li>
        <li>Last Price Increase: 2023-08-15</li>
<li>Automatic offers: </li>
<ul>
<li>hungvu193 | Contributor | 26158030</li>
<li>Nathan-Mulugeta | Reporter | 26158033</li>
</ul></li>
    </ul>
</details>

**Comment by BhuvaneshPatil** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Copy to clipboard button does not copy content of an IOU edit comment

### What is the root cause of that problem?
`messageHtml` is empty. and we are using that to get the text to be copied.
https://github.com/Expensify/App/blob/c43491adbdde7346aba6e808d3581549b9b1624c/src/pages/home/report/ContextMenu/ContextMenuActions.js#L185<img width="326" alt="Screenshot 2023-08-15 at 11 50 54 AM" src="https://github.com/Expensify/App/assets/27822551/b1da1df2-fe57-4925-a4c4-62302aadc9c5">

### What changes do you think we should make in order to solve the problem?
currently we are using empty string as default value in case `html` is not present.
We can modify it to something like - 
```js
const messageHtml = lodashGet(message, 'html', reportAction.actionName === CONST.REPORT.ACTIONS.TYPE.MODIFIEDEXPENSE ? lodashGet(message, 'text', '') : '');
```

With this we can check if task action type is modified, and assign the value from message text.
We can definitely handle the condition someplace above if we want cleaner version.

### What alternative solutions did you explore? (Optional)

**Comment by BhuvaneshPatil** [comment]
The link to github issue shall be posted in slack thread!

**Comment by pecanoro** [comment]
Discussing internally since it's a new feature!

**Comment by mountiny** [comment]
I think this needs to be fixed, but its not a deploy blocker.

**Comment by mountiny** [comment]
Its not a deploy blocker and I would expect the behaviour to be copying what you can see on the line

**Comment by pecanoro** [comment]
@0xmiroslav I can't reproduce the issue you linked so I closed it since it's pretty old.

Regarding this issue, I talked to @mountiny and we decided not to block on this and have a contributor fix it!

**Comment by pecanoro** [comment]
Heads up! After discussing it in the other issue, we think we should also fix the copy functionality for the missing previews (these ones [here](https://github.com/Expensify/App/issues/22387#issuecomment-1678854815)) and make it work the same way as this one will work.

With that in mind, I suggest updating the proposals!

**Comment by pecanoro** [comment]
@dukenv0307 All of them!

**Comment by narefyev91** [comment]
Proposal from @hungvu193 looks good to me https://github.com/Expensify/App/issues/24569#issuecomment-1678741285
🎀 👀 🎀 C+ reviewed

**Comment by narefyev91** [comment]
> 

Hey @dukenv0307 - not sure what to check here - the root case of the issue and correct fix for the issue was correctly proposed by @hungvu193 even before @pecanoro add more pages in which fixes should be applied - any specific code changes will be checked and discussed during PR review

**Comment by narefyev91** [comment]
> @narefyev91 @pecanoro suggests updating proposal here [#24569 (comment)](https://github.com/Expensify/App/issues/24569#issuecomment-1678885305). And then with hungvu proposal, it doesn't show copy option for `TaskPreview` that doesn't expected now.
> 
> > We should also update our shouldShow condition of copyToClipboard to make it show with ReportPreview
> 
> And this line in the proposal is wrong because `ReportPreview` already has copy option and we should update the condition to make copy option show it `IOUPreview` and `TaskPreview` [#24569 (comment)](https://github.com/Expensify/App/issues/24569#issuecomment-1678915222)

I'm not really see why you think that after this line "We should also update our shouldShow condition of copyToClipboard to make it show with ReportPreview" - proposal is wrong? To make it working we should update shouldShow function, am i right @dukenv0307 - again we do not checking the code lines in proposals (for that we have PR's) - the main idea is to identify the root case, identify correct places in which we should fix an issue. In any case - final decision will be made by @pecanoro

**Comment by BhuvaneshPatil** [comment]
@narefyev91 I have updated the proposal - https://github.com/Expensify/App/issues/24569#issuecomment-1678755779

It handles the case for TaskPreview correctly and which no one has mentioned about, for `TaskPreview` we need get text differently. 
Please have a look.

**Comment by pecanoro** [comment]
I agree with @narefyev91 here, it's basically the same solution and it seems @hungvu193 was faster, so I am going to assign them to the issue.

**Comment by flaviadefaria** [comment]
Payment summary:

>@hungvu193 requires payment [offer](https://www.upwork.com/nx/wm/offer/26158030) (Contributor)
>@narefyev91 does not require payment (Contractor)
>@Nathan-Mulugeta requires payment [offer](https://www.upwork.com/nx/wm/offer/26158033) (Reporter)

@Nathan-Mulugeta = $250 reporting bonus
@hungvu193 = $1000 (no 50% bonus)

@narefyev91 can you please fill out the checklist above?

**Comment by narefyev91** [comment]
- The PR that introduced the bug has been identified. Link to the PR - no linked PR because it's a new implementation - no one implement fix for that flow
- The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: N/A
- A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: N/A
- Determine if we should create a regression test for this bug - N/A

**Comment by pecanoro** [comment]
Sorry! Yes, it seems like a regression, but only one regression as it's basically the same bug with the same cause. Let's fix it so it copies the updated field.

**Comment by flaviadefaria** [comment]
Alright payment is on hold until the regression has been fixed.

**Comment by narefyev91** [comment]
> #26571

To be honest the changes should be done on BE side and correct string should come from it, and as I see from this comment https://github.com/Expensify/App/issues/26621#issuecomment-1715050617 - BE team will implement localisation on their side - and on FE we will avoid all this tricks to make it working correctly. But if we come back to initial issue - i agree that fix was not covering all possible cases

**Comment by narefyev91** [comment]
> I am still waiting on some sort of confirmation from the author or C+ that the linked issues will be solved as regressions.

To be honest the changes should be done on BE side and correct string should come from it, and as I see from this comment https://github.com/Expensify/App/issues/26621#issuecomment-1715050617 - BE team will implement localisation on their side - and on FE we will avoid all this tricks to make it working correctly. But if we come back to initial issue - i agree that fix was not covering all possible cases

**Comment by pecanoro** [comment]
@narefyev91 That's a different issue, the translations are not related to the regression and should not be considered as such since it's the BE. Rajat linked another issue https://github.com/Expensify/App/issues/26571.

@flaviadefaria Let's pay the issue with one regression, the other contributor is fixing them in their PR.

**Comment by flaviadefaria** [comment]
>@hungvu193 requires payment [offer](https://www.upwork.com/nx/wm/offer/26158030) (Contributor)
@narefyev91 does not require payment (Contractor)
@Nathan-Mulugeta requires payment [offer](https://www.upwork.com/nx/wm/offer/26158033) (Reporter)

Payment Summary:

@Nathan-Mulugeta = $250 (Reporting bonus)
@hungvu193 = $1000 (no 50% bonus) - 50% (regression) = $500

**Comment by flaviadefaria** [comment]
Everyone has been paid and since the regression is being fixed in another issue I'll close this one.
